src/python/pants/backend/python/goals/lockfile_test.py

```diff
@@ -3,20 +3,76 @@
 
 from __future__ import annotations
 
+import json
 from textwrap import dedent
 
 from pants.backend.python.goals.lockfile import (
     GeneratePythonLockfile,
     RequestedPythonUserResolveNames,
-    setup_user_lockfile_requests,
 )
+from pants.backend.python.goals.lockfile import rules as lockfile_rules
+from pants.backend.python.goals.lockfile import setup_user_lockfile_requests
 from pants.backend.python.subsystems.setup import PythonSetup
 from pants.backend.python.target_types import PythonRequirementTarget
+from pants.backend.python.util_rules import pex
 from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
-from pants.core.goals.generate_lockfiles import UserGenerateLockfiles
+from pants.core.goals.generate_lockfiles import GenerateLockfileResult, UserGenerateLockfiles
+from pants.engine.fs import DigestContents
 from pants.engine.rules import SubsystemRule
 from pants.testutil.rule_runner import PYTHON_BOOTSTRAP_ENV, QueryRule, RuleRunner
 from pants.util.ordered_set import FrozenOrderedSet
+from pants.util.strutil import strip_prefix
+
+
+def test_lockfile_generation() -> None:
+    rule_runner = RuleRunner(
+        rules=[
+            *lockfile_rules(),
+            *pex.rules(),
+            QueryRule(GenerateLockfileResult, [GeneratePythonLockfile]),
+        ]
+    )
+    rule_runner.set_options([], env_inherit=PYTHON_BOOTSTRAP_ENV)
+    result = rule_runner.request(
+        GenerateLockfileResult,
+        [
+            GeneratePythonLockfile(
+                requirements=FrozenOrderedSet(["ansicolors==1.1.8"]),
+                interpreter_constraints=InterpreterConstraints(),
+                resolve_name="test",
+                lockfile_dest="test.lock",
+                use_pex=True,
+            )
+        ],
+    )
+
+    digest_contents = rule_runner.request(DigestContents, [result.digest])
+    assert len(digest_contents) == 1
+    lock_content = digest_contents[0].content.decode()
+    header = dedent(
+        """\
+        # This lockfile was autogenerated by Pants. To regenerate, run:
```

Comment:
Does Pants pre-strip this stuff before handing to the tool to resolve with later? With a json lock file it will have to. With any lock file really. It's probably actually an error to be adding an extension to the file since Pants should not even know what the contents of those files are - and thus can't know if the comment style is corrupting of the file or not.

Reply:
It does not currently -- we'll have to strip when hooking up Pex consumption, I suppose.

Would you be willing to consider changing PEX to use TOML? We recently changed Pants's proprietary JVM lock format to use TOML, in part to accommodate comments and in part because it's arguably easier to read (less nesting). I imagine a major turnoff would be needing to vendor `toml` or `tomli`.

Reply:
See https://github.com/pantsbuild/pants/issues/14281. Feedback welcomed :)